project-management/src/main/java/life/qbic/projectmanagement/application/measurement/validation/MeasurementNGSValidator.java

```diff
@@ -52,12 +48,13 @@ public static boolean isNGS(Collection<String> properties) {
     if (properties.isEmpty()) {
       return false;
     }
-    if (properties.size() != NGS_PROPERTIES.size()) {
+    if (properties.size() != NGS_PROPERTY.values().length) {
       return false;
     }
-    for (String ngsProperty : NGS_PROPERTIES) {
+    for (NGS_PROPERTY ngsProperty : NGS_PROPERTY.values()) {
       var propertyFound = properties.stream()
-          .filter(property -> Objects.equals(property.toLowerCase(), ngsProperty)).findAny();
+          .filter(property -> Objects.equals(property.toLowerCase(), ngsProperty.label()))
+          .findAny();
       if (propertyFound.isEmpty()) {
         return false;
       }
```

Comment:
You are going through a list and checking whether this list contains all elements, correct?
Why not map the enum values to the required labels and call containsAll instead? This way java handles the optimisation of the search.

Reply:
Good point, IntelliJ recommended to wrap the thing in a hashset to avoid performance issues which was news to me 😅